### PR TITLE
Add more docs to repl completion.

### PR DESCRIPTION
**Pull request by dhoegh**

### CODE REVIEW

File: doc/manual/interacting-with-julia.rst [hunk 1]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
     \hat              \heartsuit         \hksearow          \hookleftarrow     \hslash
     \hbar             \hermitconjmatrix  \hkswarow          \hookrightarrow    \hspace
 
+    julia> α="\alpha[TAB]"   # LaTeX complation do also work in strings
```
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: completion also works in strings

File: doc/manual/interacting-with-julia.rst [hunk 2]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+    julia> path="/[TAB]"
+    .dockerenv  .juliabox/   boot/        etc/         lib/         media/       opt/         root/        sbin/        sys/         usr/
+    .dockerinit bin/         dev/         home/        lib64/       mnt/         proc/        run/         srv/         tmp/         var/
+    shell> /[TAB]
+    .dockerenv  .juliabox/   boot/        etc/         lib/         media/       opt/         root/        sbin/        sys/         usr/
+    .dockerinit bin/         dev/         home/        lib64/       mnt/         proc/        run/         srv/         tmp/         var/
+
+Tab completion can help with investigation of the available methods matching the input arguments::
+
+    julia> max([TAB] # All methods is displayed, not display here due to size of the list
```
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: methods are displayed, not shown here ...

File: doc/manual/interacting-with-julia.rst [hunk 3]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+    max{T1<:Real,T2<:Real}(x::AbstractArray{T1,N<:Any}, y::AbstractArray{T2,N<:Any}) at operators.jl:548
+    max(x, y) at operators.jl:78
+    max(a, b, c, xs...) at operators.jl:119
+
+    julia> max([1,2], max(1,2),[TAB] # All methods matching the arguments.
+    max{T1<:Real,T2<:Real}(x::AbstractArray{T1,N<:Any}, y::T2) at operators.jl:544
+    max(x, y) at operators.jl:78
+    max(a, b, c, xs...) at operators.jl:119
+
+    julia> split("1 1 1", # Keywords are also displayed in the suggested methods after the `;`
```
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: are you referring to the `;` below, or should this be a `;` on the tab-completion line to get keywords to show up?

File: doc/manual/interacting-with-julia.rst [hunk 4]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+    split(str::AbstractString) at strings/util.jl:151
+    split{T<:AbstractString}(str::T, splitter; limit, keep) at strings/util.jl:127
+
+The completion of the methods uses type inference and can therefore see if the arguments match even if the arguments are output from functions. The function needs to be type stable for the completion to be able remove non-matching methods.
```
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: to be able to remove

File: doc/manual/interacting-with-julia.rst [hunk 5]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+
+    julia> Pkg.a
+    add       available
+
+Fields for output from functions can also be completed::
+
+    julia> split("","")[1].[TAB]
+    endof  offset  string
+
+The completion of fields for output from functions uses type inference and it can only suggest fields, if the function is type stable.
```
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: move the comma to before the "and"

File: doc/manual/interacting-with-julia.rst (hunk 2 above)
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: the "all" was fine

Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: and so was the "due to size of the list"

File: doc/manual/interacting-with-julia.rst [hunk 6]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+    .dockerinit bin/         dev/         home/        lib64/       mnt/         proc/        run/         srv/         tmp/         var/
+    shell> /[TAB]
+    .dockerenv  .juliabox/   boot/        etc/         lib/         media/       opt/         root/        sbin/        sys/         usr/
+    .dockerinit bin/         dev/         home/        lib64/       mnt/         proc/        run/         srv/         tmp/         var/
+
+Tab completion can help with investigation of the available methods matching the input arguments::
+
+    julia> max([TAB] # Methods are displayed, not shown here ...
+
+    julia> max([1,2],[TAB] # All methods where array match as first argument
```
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: where array matches as

**Comment by dhoegh** [comment]
Thank you Tony for the review.

**Comment by tkelman** [comment]
`julia> max([TAB] # All methods is displayed, not display here due to size of the list` should have been left mostly intact, just as
`julia> max([TAB] # All methods are displayed, not shown here due to size of the list`, I didn't mean the "..." literally

**Comment by dhoegh** [comment]
I see. I have fixed it.

**Comment by ViralBShah** [comment]
@tkelman Merge?

**Comment by tkelman** [comment]
Usually would rather leave things open for review longer than 5 hours.

File: doc/manual/interacting-with-julia.rst [hunk 7]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+    .dockerinit bin/         dev/         home/        lib64/       mnt/         proc/        run/         srv/         tmp/         var/
+    shell> /[TAB]
+    .dockerenv  .juliabox/   boot/        etc/         lib/         media/       opt/         root/        sbin/        sys/         usr/
+    .dockerinit bin/         dev/         home/        lib64/       mnt/         proc/        run/         srv/         tmp/         var/
+
+Tab completion can help with investigation of the available methods matching the input arguments::
+
+    julia> max([TAB] # All methods are displayed, not shown here due to size of the list
+
+    julia> max([1,2],[TAB] # All methods where array matches as first argument
```
Comment (KristofferC) [inline doc/manual/interacting-with-julia.rst]: maybe be more specific and say "...where `Vector{Int}` matches..."?

File: doc/manual/interacting-with-julia.rst [hunk 8]
```diff
@@ -228,11 +228,56 @@ and get a list of LaTeX matches as well::
...
+
+    julia> split("1 1 1", # Keywords are also displayed in the suggested methods, see second line after `;` where `limit` and `keep` are keyword arguments
+    split(str::AbstractString) at strings/util.jl:151
+    split{T<:AbstractString}(str::T, splitter; limit, keep) at strings/util.jl:127
+
+The completion of the methods uses type inference and can therefore see if the arguments match even if the arguments are output from functions. The function needs to be type stable for the completion to be able to remove non-matching methods.
+
+Tab completion can also help completing fields::
+
+    julia> Pkg.a
```
Comment (KristofferC) [inline doc/manual/interacting-with-julia.rst]: Do we typically call a function in a module for a `field`?

Comment (dhoegh) [inline doc/manual/interacting-with-julia.rst]: I do not know, but `Pkg,.add` operation is lowered to `getfield(Pkg,:add)`

File: doc/manual/interacting-with-julia.rst (hunk 7 above)
Comment (dhoegh) [inline doc/manual/interacting-with-julia.rst]: I have  updated it to `Vector{Int}`.

File: doc/manual/interacting-with-julia.rst (hunk 8 above)
Comment (tkelman) [inline doc/manual/interacting-with-julia.rst]: good question whether there's a better name for module children than "field," maybe not.

**Pull request merged**